src/main/java/frc/team4909/robot/RobotConstants.java

```diff
@@ -12,8 +12,12 @@
 
   /* Drivetrain */
   //TODO: Tune values
-  public static final double speedMultiplier = .7  - 0.00012*(Math.abs(Robot.elevatorSubsystem.getPosition())); // As height increases, decrease speed
-  public static final double speedTurnMultiplier = .5 - 0.00008 * (Math.abs(Robot.elevatorSubsystem.getPosition())); // As height increases, decrease speed
+
+  public static final double speedMultiplier = .7  - ((0.7 - 0.1) / 5000) * (Math.abs(Robot.elevatorSubsystem.getPosition())); 
```

Comment:
Given that these are final the value can NEVER change. Also the code is not automatically going to re-evaluate itself (removing final doesn't solve the problem).

In a periodic() function somewhere you should recalculate the multipliers. I recommend you make a static method in `RobotConstants` that can then be called from `robot.java` `robotPerodic()` to update the multipliers.

Reply:
I would recommend that rather than have a comment describing all of the numbers in the formula, make private variables with descriptive names for each.

```java
private static final bottomSpeedMultiplier = .7;
private static final topSpeedMultiplier = .7;
private static final elevatorEncoderTicks = 5000; //is total height in encoder ticks

public static double speedMultiplier = 0; //note: Value will be set by updateSpeedMultiplier called periodically from robotPeriodic()
public static void updateSpeedMultiplier() {
    speedMultiplier = bottomSpeedMultiplier  - ((bottomSpeedMultiplier - topSpeedMultiplier) / elevatorEncoderTicks) * (Math.abs(Robot.elevatorSubsystem.getPosition()));
}


```

Reply:
I see. My bad, will fix now.

Reply:
I changed it so the calculation would be done by in drive train subsystem, removing the need to call an updateSpeedMultiplier